product/synquacer/module/synquacer_system/src/load_secure_fw.c

```diff
@@ -69,55 +74,56 @@ typedef struct arm_tf_fip_package_s arm_tf_fip_package_t;
  * Current implementation expects bl32 is located as 4th binary
  * in the arm-tf fip package.
  */
-static void fw_fip_load_bl32(void)
+static void fw_fip_load_bl32(arm_tf_fip_package_t *fip_package_p)
 {
     uint32_t trans_addr_39_20;
-    void *src, *dst;
-    uint8_t bl32_uuid[] = UUID_SECURE_PAYLOAD_BL32;
-
-    arm_tf_fip_package_t *fip_package_p =
-        (arm_tf_fip_package_t *)CONFIG_SCB_ARM_TF_BASE_ADDR;
-
-    if (memcmp(
-            (void *)bl32_uuid,
-            (void *)fip_package_p->fip_toc_entry[BL32_TOC_ENTRY_INDEX].uuid,
-            sizeof(bl32_uuid)) != 0) {
-        FWK_LOG_ERR("[FIP] BL32 UUID is wrong, skip loading");
-        return;
-    }
-
-    FWK_LOG_ERR("[FIP] load BL32");
+    uint32_t src;
+    void *dst;
 
     /* enable DRAM access by configuring address trans register */
     trans_addr_39_20 =
         ((CONFIG_SCB_ARM_TB_BL32_BASE_ADDR >> 20) | ADDR_TRANS_EN);
     *((volatile uint32_t *)(REG_ASH_SCP_POW_CTL + ADDR_TRANS_OFFSET)) =
         trans_addr_39_20;
 
-    src = (void *)((uint32_t)fip_package_p +
-     (uint32_t)fip_package_p->fip_toc_entry[BL32_TOC_ENTRY_INDEX].offset_addr);
+    src = CONFIG_SCB_ARM_TF_OFFSET +
+        (uint32_t)fip_package_p->fip_toc_entry[BL32_TOC_ENTRY_INDEX]
+            .offset_addr;
     dst = (void *)SCP_ADDR_TRANS_AREA;
-    memcpy(dst, src, fip_package_p->fip_toc_entry[BL32_TOC_ENTRY_INDEX].size);
+
+    FWK_LOG_INFO(
+        "[FIP] load BL32 src 0x%08lx, dst %08lx, size 0x%lx",
```

Comment:
This is failing the build the build with the error:
```
scp/product/synquacer/module/synquacer_system/src/load_secure_fw.c:96:9: warning: format specifies type 'unsigned long' but the argument has type 'uint32_t' (aka 'unsigned int') [-Wformat]
        src,
```
Can you please change it to:
```
--- a/product/synquacer/module/synquacer_system/src/load_secure_fw.c
+++ b/product/synquacer/module/synquacer_system/src/load_secure_fw.c
@@ -92,7 +92,7 @@ static void fw_fip_load_bl32(arm_tf_fip_package_t *fip_package_p)
     dst = (void *)SCP_ADDR_TRANS_AREA;
 
     FWK_LOG_INFO(
-        "[FIP] load BL32 src 0x%08lx, dst %08lx, size 0x%lx",
+        "[FIP] load BL32 src 0x%08x, dst %08x, size 0x%x",
         src,
         (uint32_t)dst,
         (uint32_t)fip_package_p->fip_toc_entry[BL32_TOC_ENTRY_INDEX].size);
```

Thanks

Reply:
@tarek-arm 
This log is duplicate so I will remove this line.

Reply:
@tarek-arm
I found problem of loading bl32, I fixed the process of loading bl32 and I still need this log output.
In my environment, I use gcc-arm-none-eabi-10.3-2021.07 for cross compiler and I encounter build error when I change from "0x%08lx" to "0x%08x". I will use PRIx32/PRIu32 instead in the next version.